go-controller/pkg/ovn/controller/apbroute/repair.go

```diff
@@ -395,3 +417,63 @@ func (c *ExternalGatewayMasterController) buildOVNECMPCache() (map[string][]*ovn
 	}
 	return ovnRouteCache, nil
 }
+
+// returns a hybrid cache of map[podIPs]nodeName
+func (c *ExternalGatewayMasterController) buildOVNHybridCache() (map[string]string, error) {
+	ovnHybridCache := make(map[string]string)
+	p := func(item *nbdb.LogicalRouterPolicy) bool {
+		return item.Priority == types.HybridOverlayReroutePriority
+	}
+
+	logicalRouterPolicies, err := c.nbClient.findLogicalRouterPoliciesWithPredicate(p)
+	if err != nil {
+		return nil, fmt.Errorf("CleanHybridPRoutes: failed to list hybrid routes: %v", err)
+	}
+
+	foundNextHops := sets.Set[string]{}
+	for _, lrp := range logicalRouterPolicies {
+		foundNextHops.Insert(lrp.Nexthops...)
+	}
+
+	r := func(item *nbdb.LogicalRouterPort) bool {
+		for _, ip := range item.Networks {
+			// grab only IP prefix and not mask
+			p := strings.Split(ip, "/")
+			if foundNextHops.Has(p[0]) {
+				return true
+			}
+		}
+		return false
+	}
+
+	grPorts, err := c.nbClient.findLogicalRouterPortWithPredicate(r)
```

Comment:
ah this is being done to get the nodeName for the address-set huh.. for IC it would be so much easier since zoneName is nodeName and we wouldn't have to care with the extra logic, but for nonIC with globalDB we need this....